src/main/java/com/nedap/openehr/lsp/paths/PathUtils.java

```diff
@@ -0,0 +1,129 @@
+package com.nedap.openehr.lsp.paths;
+
+import com.google.common.collect.Lists;
+import com.nedap.archie.aom.Archetype;
+import com.nedap.archie.aom.ArchetypeModelObject;
+import com.nedap.archie.aom.CAttribute;
+import com.nedap.archie.aom.CObject;
+import com.nedap.archie.archetypevalidator.ValidationResult;
+import com.nedap.archie.paths.PathSegment;
+import com.nedap.archie.rminfo.MetaModels;
+import com.nedap.openehr.lsp.ADL2TextDocumentService;
+import com.nedap.openehr.lsp.document.HoverInfo;
+import org.eclipse.lsp4j.CodeLens;
+import org.eclipse.lsp4j.Command;
+import org.eclipse.lsp4j.Hover;
+import org.eclipse.lsp4j.MarkupContent;
+import org.openehr.bmm.core.BmmClass;
+import org.openehr.bmm.core.BmmProperty;
+import org.openehr.bmm.persistence.validation.BmmDefinitions;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class PathUtils {
+
+    public static void createCodeLenses(List<CodeLens> result, ArchetypePathReference reference, ValidationResult validationResult) {
+        Archetype flattened = validationResult.getFlattened();
+        if (flattened != null) {
+            PartialAOMPathQuery aomPathQuery = new PartialAOMPathQuery(reference.getPath());
+            PartialAOMPathQuery.PartialMatch partial = aomPathQuery.findLSPPartial(flattened.getDefinition());
+            if (partial.getMatches().size() > 0) {
+                ArchetypeModelObject archetypeModelObject = partial.getMatches().get(0);
+                String content = null;
+                String description = null;
+                if(archetypeModelObject instanceof CAttribute) {
+                    content = findNearestText((CAttribute) archetypeModelObject);
+                    description = findNearestDescription((CAttribute) archetypeModelObject);
+                } else if (archetypeModelObject instanceof CObject){
+                    content = findNearestText((CObject) archetypeModelObject);
+                    description = findNearestDescription((CObject) archetypeModelObject);
+                }
+                String text = content + partial.getRemainingQuery().stream().map(PathSegment::toString).collect(Collectors.joining("/"));
+                String extraText = description;
+                extraText += "\n\nIn Archetype " + flattened.getDefinition().getTerm().getText() + " (" + reference.getArchetypeId() +")";
+                if(reference.getExtraInformation() != null) {
+                    text += " " + reference.getExtraInformation();
+                }
+
+                //result.add(new CodeLens(reference.getRange(), new Command(text, ADL2TextDocumentService.SHOW_INFO_COMMAND, Lists.newArrayList(extraText)), null));
+                result.add(new CodeLens(reference.getRange(), new Command(text, ADL2TextDocumentService.SHOW_INFO_COMMAND, Lists.newArrayList(extraText)), null));
+            } else {
```

Comment:
Empty else statements and commented code

Reply:
The results.add I will remove. I think the empty else and potential hover add something: it indicates the easy possibility of adding that feature if we decide it's actually better than not having it.